converter/sfxConverter.go

```diff
@@ -452,16 +469,170 @@ func NewCadvisorCollector(infoProvider infoProvider, f ContainerNameToLabelsFunc
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Cpu.Limit))}}
+			},
+		},
+	}
+
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
+			metrics[index] = metric
+			index++
+		}
+	}
+	// trim metrics down to the desired length
+	metrics = metrics[:index]
+	return metrics
+}
+
+func getContainerSpecMemMetrics(excludedMetrics map[string]bool) []containerSpecMetric {
+	var metrics = []containerSpecMetric{
+		{
+			containerMetric: containerMetric{
+				name:        "container_spec_memory_limit_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Memory.Limit))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "container_spec_memory_swap_limit_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(container *info.ContainerInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(container.Spec.Memory.SwapLimit))}}
+			},
+		},
+	}
+
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
+			metrics[index] = metric
+			index++
+		}
+	}
+	// trim metrics down to the desired length
+	metrics = metrics[:index]
+	return metrics
+}
+
+func getMachineInfoMetrics(excludedMetrics map[string]bool) []machineInfoMetric {
+	var metrics = []machineInfoMetric{
+		{
+			containerMetric: containerMetric{
+				name:        "machine_cpu_frequency_khz",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.CpuFrequency))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "machine_cpu_cores",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.NumCores))}}
+			},
+		},
+		{
+			containerMetric: containerMetric{
+				name:        "machine_memory_bytes",
+				help:        "",
+				valueType:   datapoint.Gauge,
+				extraLabels: []string{},
+			},
+			getValues: func(machineInfo *info.MachineInfo) metricValues {
+				return metricValues{{value: datapoint.NewIntValue(int64(machineInfo.MemoryCapacity))}}
+			},
+		},
+	}
+	var index = 0
+	for _, metric := range metrics {
+		// check if metric is not on the exclusion list
+		if _, ok := excludedMetrics[metric.name]; !ok {
```

Comment:
Trying to figure out how this works. If metrics[0] is not excluded then we check metrics[1] and it is excluded is index not still equal to 0 and will overwrite metrics[0]?